x/profile/internal/keeper/querier.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/desmos-labs/desmos/x/profile/internal/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// NewQuerier is the module level router for state queries
+func NewQuerier(keeper Keeper) sdk.Querier {
+	return func(ctx sdk.Context, path []string, req abci.RequestQuery) (res []byte, err error) {
+		switch path[0] {
+		case types.QueryProfile:
+			return queryProfile(ctx, path[1:], req, keeper)
+		case types.QueryProfiles:
+			return queryProfiles(ctx, req, keeper)
+		default:
+			return nil, fmt.Errorf("unknown post query endpoint")
+		}
+	}
+}
+
+// queryProfile handles the request to get a profile having a moniker
+func queryProfile(ctx sdk.Context, path []string, _ abci.RequestQuery, keeper Keeper) ([]byte, error) {
+	address, err := sdk.AccAddressFromBech32(path[0])
```

Comment:
Probably we should also allow to query using the moniker, so if this fails we should use `path[0]` as a moniker instead.